mysql/tests/test_mysql.py

```diff
@@ -217,6 +217,30 @@ def _test_optional_metrics(aggregator, optional_metrics, at_least):
     assert before - after > at_least
 
 
+@pytest.mark.unit
+def test_innodb_status_unicode_error():
+    mysql_check = MySql(common.CHECK_NAME, {}, instances=[{}])
+    mysql_check.log = mock.MagicMock()
+
+    class MockCursor:
+        def execute(self, command):
+            raise UnicodeDecodeError('encoding', b'object', 0, 1, command)
+
+        def close(self):
+            return MockCursor()
+
+    class MockDatabase:
+        def cursor(self):
+            return MockCursor()
+
+    assert mysql_check._get_stats_from_innodb_status(MockDatabase()) == {}
+    mysql_check.log.warning.assert_called_with(
+        "Unicode error while getting INNODB status "
+        "(if this warning is infrequent, metric collection won't be impacted): %s",
+        "'encoding' codec can't decode byte 0x6f in position 0: SHOW /*!50000 ENGINE*/ INNODB STATUS",
+    )
```

Comment:
(Nit) Non-mocking alternative using [caplog](https://docs.pytest.org/en/stable/logging.html#caplog-fixture) that also doesn't require writing the entire warning message (could be brittle to maintain):

```python
with caplog.at_level(logging.WARNING):
    assert mysql_check._get_stats_from_innodb_status(MockDatabase()) == {}

for _, _, message in caplog.record_tuples:
    if 'Unicode error while getting INNODB status' in message:
        break
else:
    pytest.fail('Warning message not found')
```

Or, even simpler:

```python
caplog.set_level(logging.WARNING)
assert mysql_check._get_stats_from_innodb_status(MockDatabase()) == {}
assert 'Unicode error while getting INNODB status' in message in caplog.text
```

(We could then drop the `log` mock.)